src/main/java/slimeknights/tconstruct/library/recipe/casting/PartCastingRecipe.java

```diff
@@ -0,0 +1,55 @@
+package slimeknights.tconstruct.library.recipe.casting;
+
+import net.minecraft.fluid.Fluids;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.library.materials.IMaterial;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.material.MaterialCastingRecipe;
+import slimeknights.tconstruct.library.tinkering.IMaterialItem;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+
+public class PartCastingRecipe extends AbstractCastingRecipe {
```

Comment:
Don't extend `AbstractCastingRecipe`, thats a poor fit. Make a new `ICastingRecipe` an have both classes implement. Removes the need to change `AbstractCastingRecipe` for the sake of a dynamic recipe.

Alternatively, put out the common logic between abstract and part casting into a shared parent. The key detail is I dislike that you pass in an empty fluid to meet the need of the parent (its going to be a fluid ingredient in the future anyways, so we will have to break this again)